### PR TITLE
Moddable Lua Needs

### PR DESCRIPTION
**Pull request by Dan54**
* Makes needs capable of using Lua. C# need handling is still implemented, but can be disabled with ```<LuaProcessingOnly />```. 
* ```<OnUpdate>function_name</OnUpdate>``` adds a function to be run every frame, taking (need, deltaTime) arguments.
* Needs now support mods.

### CODE REVIEW

**Comment by svmnotn** [comment]
Did this pass StyleCop?

File: Assets/Scripts/Models/NeedActions.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        myLuaScript.Globals["Job"] = typeof(Job);
+
+        // Also to access statics/globals
+        myLuaScript.Globals["World"] = typeof(World);
+    }
+
+    public static void addScript(string rawLuaCode)
+    {
+        _Instance.myLuaScript.DoString(rawLuaCode);
+    }
```
Comment (abackwood) [inline Assets/Scripts/Models/NeedActions.cs]: Method names should start with Capitals

File: Assets/Scripts/Models/NeedActions.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    
+	public static void CallFunctionsWithNeed(string[] functionNames, Need need, float deltaTime)
+    {
```
Comment (abackwood) [inline Assets/Scripts/Models/NeedActions.cs]: Replace tabs with sapces

**Comment by abackwood** [comment]
You used tabs instead of spaces all over the code

**Comment by Dan54** [comment]
Sorry, project policies messed up.

**Comment by abackwood** [comment]
Where is `<LuaProcessingOnly/>`? I can't find it in Settings.xml

**Comment by Dan54** [comment]
@abackwood it's in Need.ReadXmlPrototype()

**Comment by abackwood** [comment]
Seems to work and indeed doesn't process when LueProcessingOnly is enabled for Oxygen.
Haven't tested the OnUpdate functionality (yet).
Still some style issues. Run StyleCop

That being said: great initiative! I expect I'll be using this when it gets merged :)

**Comment by Mizipzor** [comment]
Tested. I started the game and the workers oxygen need was going down. Thats good enough since the needs system is a bit up in the air. Do we have a good way to test it?

**Pull request merged**